cumulus/client/consensus/aura/src/collators/lookahead.rs

```diff
@@ -315,13 +315,14 @@ where
 			let mut parent_header = initial_parent.header;
 			let overseer_handle = &mut params.overseer_handle;
 
-			// We mainly call this to inform users at genesis if there is a mismatch with the
-			// on-chain data.
-			collator.collator_service().check_block_status(parent_hash, &parent_header);
```

Comment:
Could we not directly call `continue` here? Why move it to the loop below?
Parent hash and parent header below will only change if we built a new block outselves, so that should always return true in subsequent calls anyway (or  I am missing something).

Reply:
`parent_hash` and `parent_header` get updated at the bottom of the inner loop. On the other hand, they are updated to values we get after a successful collation, which are unlikely to be unknown or bad  :thinking: Need to think a bit more, thanks for pointing out!

Reply:
> `parent_hash` and `parent_header` get updated at the bottom of the inner loop. On the other hand, they are updated to values we get after a successful collation, which are unlikely to be unknown or bad 🤔 Need to think a bit more, thanks for pointing out!

Exactly, that is what I meant. We build a block successfully and set parent to it. But since we built it ourselves it will for sure be in-chain. If the first parent itself if messed up we will abort before entering the `0..2` loop.